include/boost/url/string.hpp

```diff
@@ -55,38 +104,117 @@ class string_value : public string_view
         char*& dest);
 
 public:
+    /** The default allocator type to construct read-only instances of @ref string_value
+
+        An allocator of this class can be instantiated to create
+        multiple read-only string values.
+
+        @note This custom allocator is not an instance of the C++ Allocator named requirement
+
+        @par References
+        @li <a href="https://en.cppreference.com/w/cpp/named_req/Allocator">Allocator (cppreference)</a>
+
+     */
     class allocator;
 
+    /** Destructor
+
+        Upon destruction, ownership of the shared string is released. If
+        there are no more owners, the resources for the string are released.
+
+     */
     inline
     ~string_value();
 
+    /** Default constructor
+
+        This function creates the default base @ref string_view, which points to
+        no string / nullptr.
+
+        @note As an optimization, the reference counter is `nullptr`.
+
+     */
     string_value() = default;
 
+    /** Construct an initial string_value
+
+        This function creates a string of the specified size at the
+        destination with the a custom allocator.
+
+        @note Although this function returns sets the destination pointer
+        to the new string address, this string_value owns the buffer.
+        The buffer is destroyed after the last string_value points to it.
+
+        @tparam Allocator Allocator type
```

Comment:
you don't need to document the tparam for a well known concept. Try changing the declaration to say `__Allocator__` and see if docca turns it into a link.